compiler/damlc/daml-doc/src/DA/Daml/Doc/Extract.hs

```diff
@@ -497,28 +500,29 @@ getTemplateDocs DocCtx{..} typeMap templateInstanceMap =
                       [] -> [] -- catching the dummy case here, see above
                       _other -> error "getFields: found multiple constructors"
 
--- | A template instance is desugared into a newtype with a docs marker.
+-- | A template instance is desugared to a type synonym with a doc decl
+-- around it.
+--
 -- For example,
 --
 -- @template instance ProposalIou = Proposal Iou@
 --
 -- becomes
 --
--- @newtype ProposalIou = ProposalIou (Proposal Iou) -- ^ TEMPLATE_INSTANCE@
+-- @--| TEMPLATE_INSTANCE@
+-- @type ProposalIou = Proposal Iou@
 --
 -- So the goal of this function is to extract the template instance doc
--- from the newtype doc if it exists.
+-- from the doc preceding the type synonym if it exists.
 getTemplateInstanceDoc :: ADTDoc -> Maybe TemplateInstanceDoc
-getTemplateInstanceDoc adt
-    | ADTDoc{..} <- adt
-    , [PrefixC{..}] <- ad_constrs
-    , Just (DocText "TEMPLATE_INSTANCE") <- ac_descr
-    , [argType] <- ac_args
+getTemplateInstanceDoc tyConDoc
+    | TypeSynDoc{..} <- tyConDoc
+    , Just (DocText "TEMPLATE_INSTANCE") <- ad_descr
```

Comment:
This looks like it will only recognise the template instance if the user does not supply any docs for it.

For example,

```Haskell
-- | This is my template instance
template instance ProposalIou = Proposal Iou
```

would desugar into something like

```Haskell
-- | This is my template instance
-- | TEMPLATE_INSTANCE
type ProposalIou = Proposal Iou
```

As a quick fix I suggest checking for `TEMPLATE_INSTANCE` inside the doctext with `isInfixOf`, and we can fix the contents of the doctext later.

Reply:
Thanks for the review! I've added something like a suffix check here to remove `"TEMPLATE_INSTANCE"` from the real documentation. I haven't seen if it works yet because there is still some work to handle type synonyms in the compiler back end.

Reply:
👍 That looks good, but definitely worth testing :-)